crates/pgsrv/src/proxy.rs

```diff
@@ -339,14 +339,21 @@ impl<A: ProxyAuthenticator> ProxyHandler<A> {
                     None => user,
                 };
 
+                // Extract the compute engine name (optional) from startup params
+                // Defaults to empty string which will target the shared pool
+                let compute_engine = match params.get("compute_engine") {
+                    Some(compute_engine) => compute_engine,
+                    None => "",
+                };
```

Comment:
We'll want to grab this from `options` (which corresponds to the options part of the connection string).